sw/cheri/checks/tag_test.S

```diff
@@ -138,6 +138,28 @@ revtags:
   cgettag a3, ct1
   beqz a3, fail
 
+// Test of hardware revoker
+hw_rev:
+  // Capability to revocation tags
+  li t0, 0x30000000
+  csetaddr ct0, ca0, t0
+  // Zero revocation tags
+  csw zero, 0(ct0)
+  // Capability to hardware revoker control
+  li t1, 0x8000A000
+  csetaddr ct1, ca0, t1
+  li t2, 0x00120000 // End of SRAM
+  addi t2, t2, -8 // Point to last capability aligned region in SRAM
+  csw t2, 4(ct1) // Set end of memory to sweep
+  addi t2, t2, -8 // Second to last capability in SRAM.
+  csw t2, 0(ct1) // Set start of memory to sweep
+  clw t2, 12(ct1) // Read current epoch
+  andi t2, t2, 1 // Get running bit
+  bnez t2, fail
+  li t2, 1
+  csw t2, 20(ct1) // Enable interrupts
+  csw t2, 8(ct1) // Start sweep
```

Comment:
Shouldn't we extend the test to ensure the revoker finishes and setup something for it to revoke? As is we're testing it doesn't report it's running if we haven't started it and that we're not immediately deadlocking the CPU by starting it off.

Reply:
Yes, at the moment I am running this test in simulation and checking the waveform to see if it does what I expect. Let me see if I can extend this a bit so that it at least tests something in case no-one looks at the waveform.

Reply:
I've at least checked the interrupt status now, but I agree we should probably add an actual revocation in here.